### PR TITLE
WIP: Basic collection models implementation

### PR DESCRIPTION
**Pull request by cutwater**

### CODE REVIEW

**Comment by alikins** [comment]
Going to try to update https://github.com/ansible/galaxy-api/pull/7 to rebase on top of this or otherwise combine them.

File: galaxy_api/api/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("name", models.CharField(editable=False, max_length=64, unique=True)),
+                ("company", models.CharField(blank=True, max_length=64)),
+                ("email", models.CharField(blank=True, max_length=256)),
+                ("logo_url", models.CharField(blank=True, max_length=256)),
```
Comment (newswangerd) [inline galaxy_api/api/migrations/0001_initial.py]: Lets use `avatar_url` here, to try and maintain better compatibility with galaxy.

File: galaxy_api/api/models.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        links: Reference to related links.
+
+    """
+
+    # Fields
+
+    name = models.CharField(max_length=const.MAX_NAME_LENGTH, unique=True, editable=False)
+    company = models.CharField(max_length=64, blank=True)
+    email = models.CharField(max_length=256, blank=True)
+    logo_url = models.CharField(max_length=256, blank=True)
```
Comment (newswangerd) [inline galaxy_api/api/models.py]: Lets use `avatar_url` here, to try and maintain better compatibility with galaxy.

Comment (cutwater) [reply galaxy_api/api/models.py]: Ok

File: galaxy_api/auth/auth.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        header = self._decode_header(request.META[self.header])
+
+        try:
+            identity = header['identity']
+            account = identity['account_number']
+
+            user = identity['user']
+            username = user['username']
+        except KeyError:
+            raise AuthenticationFailed
```
Comment (alikins) [inline galaxy_api/auth/auth.py]: Worth adding some detail on what AuthenticationFailed?

Or is it left blank without a .authenticate_header() method to cause a 403 without any extra info? (if I'm reading https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication correctly).

File: galaxy_api/api/models.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    email = models.CharField(max_length=256, blank=True)
+    avatar_url = models.CharField(max_length=256, blank=True)
+    description = models.CharField(max_length=256, blank=True)
+
+    # TODO(cutwater): Consider moving to a standalone model
+    resources_page = models.TextField(blank=True)
+    resources_page_html = models.TextField(blank=True, editable=False)
+
+    # References
+    owners = models.ManyToManyField(auth_models.Tenant, related_name='namespaces')
```
Comment (alikins) [inline galaxy_api/api/models.py]: Should this be named 'tenants' ?

**Pull request closed without merging**